index.js

```diff
@@ -90,7 +91,7 @@ module.exports = class Corestore extends EventEmitter {
         if (err && err.code !== 'ENOENT') return reject(err)
         if (err || st.size < 32 || this._overwrite) {
           const key = this.primaryKey || crypto.randomBytes(32)
-          return this._keyStorage.write(0, key, err => {
+          return this._keyStorage.write(0, this.persistentKey ? key : b4a.alloc(0), err => {
```

Comment:
dont write anything if you are not persisting. just return before instead and resolve

Reply:
Right, the issue is that we still need to lock from the storage

Reply:
Although I don't remember if without writing then the file exists anyway, maybe it does, will check later!